nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GrokParser.java

```diff
@@ -0,0 +1,243 @@
+        buffer = new byte[maxBufferSize];
+
+        try{
+            grok = Grok.create(context.getProperty(GROK_PATTERN_FILE).getValue());
+            grok.compile(context.getProperty(GROK_EXPRESSION).getValue());
+        }catch (GrokException e){
+            getLogger().error("Failed to initialize ExtractGrok due to: ", e);
+        }
+
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        final Charset charset = Charset.forName(context.getProperty(CHARACTER_SET).getValue());
+        final Map<String, String> grokResults = new HashMap<>();
+        final byte[] byteBuffer = buffer;
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(InputStream in) throws IOException {
+                StreamUtils.fillBuffer(in, byteBuffer, false);
+            }
+        });
+        final long len = Math.min(byteBuffer.length, flowFile.getSize());
+        final String contentString = new String(byteBuffer, 0, (int) len, charset);
+
+
+        final Match gm = grok.match(contentString);
+        gm.captures();
+        for(Map.Entry<String,Object> entry: gm.toMap().entrySet()){
+            if(null != entry.getValue() ) {
+                grokResults.put(entry.getKey(), entry.getValue().toString());
+            }
+        }
+
+        if (grokResults.isEmpty()) {
+            session.transfer(flowFile, REL_NO_MATCH);
+            getLogger().info("Did not match any Grok Expressions for FlowFile {}", new Object[]{flowFile});
+            return ;
+        }
+        switch (context.getProperty(DESTINATION).getValue()){
+            case DESTINATION_ATTRIBUTE:
+
+
+                flowFile = session.putAllAttributes(flowFile, grokResults);
+
+                session.getProvenanceReporter().modifyAttributes(flowFile);
+                session.transfer(flowFile, REL_MATCH);
+                getLogger().info("Matched {} Grok Expressions and added attributes to FlowFile {}", new Object[]{grokResults.size(), flowFile});
+
+                break;
+            case DESTINATION_CONTENT:
+
+                FlowFile conFlowfile = session.write(flowFile, new StreamCallback() {
+                    @Override
+                    public void process(InputStream in, OutputStream out) throws IOException {
+                        try (OutputStream outputStream = new BufferedOutputStream(out)) {
+                            outputStream.write(objectMapper.writeValueAsBytes(grokResults));
+                        }
+                    }
+                });
+                conFlowfile = session.putAttribute(conFlowfile, CoreAttributes.MIME_TYPE.key(), APPLICATION_JSON);
+                session.getProvenanceReporter().modifyContent(conFlowfile,"Replaced content with parsed Grok fields and values");
```

Comment:
I would recommend again that we include the amount of time that it took to perform the update.